File: server/routes/api/local/comments.js
const db = require('../../../models')
const CommentViewModel = require('../../../viewmodels/comment')

module.exports = {
  getCommentsForIssue: function (req, res) {
    return db.Comment.findAll({
      where: { issueId: req.params.issueId },
      include: [{
        model: db.User,
        as: 'author'
      }]
    })
      .then(comments => {
        const result = []
        for (let comment of comments) {
          result.push(CommentViewModel.createFromLocal(comment))
        }
        res.send(result)
      })
  },
  addComment: function (req, res) {
    return db.Comment.create({
      body: req.body.body,
      authorId: req.user.id,
      issueId: req.params.issueId
    }).then(comment => {
      return db.Comment.findOne({
        where: { id: comment.id },
        include: [{
          model: db.User,
          as: 'author'
        }]
      }).then(comment => res.send(comment))
    }).catch(err => res.status(502).send(err))
  }
}

File: server/repository/local/comments.py
from ._base import BaseRepo
from model import Comment
import logging


LOGGER = logging.getLogger(__name__)
BASE_COMMENT_SELECTOR = (
    'SELECT c.id, c.body, c."createdAt", c."updatedAt", '
    'u.id user_id, u.username user_username, u."displayName" "user_displayName", '
    'u."externalId" "user_externalId", u.avatar user_avatar '
    'FROM comments c INNER JOIN users u ON u.id = c."authorId" '
)


class CommentsRepository(BaseRepo):
    def get_comments_for_issue(self, issue_id):
        sql = BASE_COMMENT_SELECTOR + 'WHERE "issueId" = %(issueId)s;'
        results = self.db.fetch(sql, {'issueId': issue_id})
        return [Comment.from_db_dict(comment) for comment in results]

    def get_comments_for_issues(self, issue_ids):
        sql = BASE_COMMENT_SELECTOR + 'WHERE "issueId" = %(issueId)s;'
        results = self.db.fetch(sql, {'issueIds': issue_ids})
        comment_map = {}
        for result in results:
            comment = Comment.from_db_dict(result)
            comment_map.setdefault(comment.id, [])
            comment_map[comment.id].append(comment)
        return comment_map

    def create(self, comment, author_id):
        sql = (
            'INSERT INTO comments (body, "authorId") VALUES '
            '(%(body)s, %(author_id)s) RETURNING *;'
        )
        result = self.db.fetch_one(sql, {
            'body': comment.get('body'),
            'author_id': comment.get('author', {}).get('id'),
        })
        sql = BASE_COMMENT_SELECTOR + 'WHERE id = %(id)s;'
        result = self.db.fetch_one(sql, {'id': result['id']})
        return Comment.from_db_dict(result)
